File: PROGRAMMERS/lv0/최빈값 구하기.js
function solution(array) {
  let m = new Map();
  for (let i of array) {
    m.set(i, (m.get(i) || 0) + 1);
  }
  m = [...m].sort((a, b) => b[1] - a[1]);
  return m.length == 1 || m[0][1] > m[1][1] ? m[0][0] : -1;
}

let inps = [[1, 2, 3, 3, 3, 4], [1, 1, 2, 2], [1]];

for (let i of inps) {
  console.log(solution(i));
}

File: PROGRAMMERS/lv0/python/최빈값 구하기.py
def solution(array):
    m = {}
    for v in array:
        m[v] = m[v]+1 if v in m else 1
    m = sorted(m.items(), key=lambda x:x[1], reverse=True)
    return m[0][0] if len(m)==1 or m[0][1] > m[1][1] else -1

inps = [
    [1, 2, 3, 3, 3, 4],
    [1, 1, 2, 2],
    [1],
]

for inp in inps:
    print(solution(inp))